[Cocoa] Enable HEIC

litherum opened the pull request:
#### 8d15007530ba97cafde25c94d23e3ec6bb6dafe4
<pre>
[Cocoa] Enable HEIC
<a href="https://bugs.webkit.org/show_bug.cgi?id=257763">https://bugs.webkit.org/show_bug.cgi?id=257763</a>
rdar://110346646

Reviewed by Said Abou-Hallawa.

There are 3 reasons to enable it:
1. HEIC uses the same technology as H.265 videos, which are already enabled by default in
       many (most? almost all?) major browsers. So there is no concern about increasing
       the surface area of new technology on the web.
2. It&apos;s just a true fact that there are just a lot of HEIC images out there in the wild.
       Some of them end up on the internet. You can&apos;t stop them. It just happens.
3. Apple&apos;s devices have hardware decoders for HEIC, which make them both more performant
       and use less energy than other image formats. Running some tests, using hardware
       decoding HEIC images uses 26% less power and 13% less time than using software
       decoding of AVIF, for (roughly) equivalent quality images.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/loader/cache/CachedResourceRequest.cpp:
(WebCore::acceptHeaderValueForHEICImageResource):
(WebCore::acceptHeaderValueForImageResource):
* Source/WebCore/platform/MIMETypeRegistry.cpp:
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::defaultSupportedImageTypes):

Canonical link: <a href="https://commits.webkit.org/264971@main">https://commits.webkit.org/264971@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/66f9720b346af702b8fbda1c8ee52de30bb46fc7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9320 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9599 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9827 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10982 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/9199 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9329 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11585 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9560 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12086 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9469 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10392 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/8035 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11141 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7688 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8501 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/15947 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/7991 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8786 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8652 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11990 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/8923 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9147 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7450 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/9496 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/8361 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/8407 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/2294 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2253 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/12585 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/9737 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8910 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2390 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

shallawa reviewed (approved):
I think we need to remove the WTR setting `<!-- webkit-test-runner [ additionalSupportedImageTypes=public.heic;public.heics ] -->` from these tests:

LayoutTests/fast/images/heic-as-background-image.html
LayoutTests/fast/images/animated-heics-draw.html
LayoutTests/fast/images/animated-heics-verify.html
LayoutTests/http/tests/misc/heic-accept-header.htm

Also I think we need to make `setAdditionalSupportedImageTypes()` assert that the additional formats are not among `defaultSupportedImageTypes()`. So the clients of WebView know that they no longer need to explicitly enable HEIF when creating it.

litherum committed:
[Cocoa] Enable HEIC
https://bugs.webkit.org/show_bug.cgi?id=257763
rdar://110346646

Reviewed by Said Abou-Hallawa.

There are 3 reasons to enable it:
1. HEIC uses the same technology as H.265 videos, which are already enabled by default in
       many (most? almost all?) major browsers. So there is no concern about increasing
       the surface area of new technology on the web.
2. It's just a true fact that there are just a lot of HEIC images out there in the wild.
       Some of them end up on the internet. You can't stop them. It just happens.
3. Apple's devices have hardware decoders for HEIC, which make them both more performant
       and use less energy than other image formats. Running some tests, using hardware
       decoding HEIC images uses 26% less power and 13% less time than using software
       decoding of AVIF, for (roughly) equivalent quality images.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/loader/cache/CachedResourceRequest.cpp:
(WebCore::acceptHeaderValueForHEICImageResource):
(WebCore::acceptHeaderValueForImageResource):
* Source/WebCore/platform/MIMETypeRegistry.cpp:
* Source/WebCore/platform/graphics/cg/UTIRegistry.cpp:
(WebCore::defaultSupportedImageTypes):

Canonical link: https://commits.webkit.org/264971@main
